lib/charms/nginx_ingress_integrator/v0/nginx_route.py

```diff
@@ -0,0 +1,386 @@
+    if owasp_modsecurity_custom_rules is not None:
+        config["owasp-modsecurity-custom-rules"] = owasp_modsecurity_custom_rules
+    if path_routes is not None:
+        config["path-routes"] = path_routes
+    if retry_errors is not None:
+        config["retry-errors"] = retry_errors
+    if rewrite_target is not None:
+        config["rewrite-target"] = rewrite_target
+    if rewrite_enabled is not None:
+        config["rewrite-enabled"] = rewrite_enabled
+    if service_namespace is not None:
+        config["service-namespace"] = service_namespace
+    if session_cookie_max_age is not None:
+        config["session-cookie-max-age"] = session_cookie_max_age
+    if tls_secret_name is not None:
+        config["tls-secret-name"] = tls_secret_name
+
+    _NginxRouteRequirer(
+        charm=charm, config=config, nginx_route_relation_name=nginx_route_relation_name
+    )
+
+
+class _NginxRouteProvider(Object):
+    """Class containing the functionality for the 'provides' side of the 'nginx-route' relation.
+
+    Attrs:
+        on: nginx-route relation event describer.
+
+    Hook events observed:
+        - relation-changed
+    """
+
+    on = _NginxRouteCharmEvents()
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        nginx_route_relation_name: str = "nginx-route",
+    ):
+        """Init function for the NginxRouterProvides class.
+
+        Args:
+            charm: The charm that provides the nginx-route relation.
+            nginx_route_relation_name: Specifies the relation name of the relation handled by this
+                provider class. The relation must have the nginx-route interface.
+        """
+        # Observe the relation-changed hook event and bind
+        # self.on_relation_changed() to handle the event.
+        super().__init__(charm, nginx_route_relation_name)
+        self._charm = charm
+        self._charm.framework.observe(
+            self._charm.on[nginx_route_relation_name].relation_changed, self._on_relation_changed
+        )
+        self._charm.framework.observe(
+            self._charm.on[nginx_route_relation_name].relation_broken, self._on_relation_broken
+        )
+
+    def _on_relation_changed(self, event: RelationChangedEvent) -> None:
+        """Handle a change to the nginx-route relation.
+
+        Confirm we have the fields we expect to receive.
+
+        Args:
+            event: Event triggering the relation-changed hook for the relation.
+        """
+        # `self.unit` isn't available here, so use `self.model.unit`.
+        if not self._charm.model.unit.is_leader():
+            return
+
+        relation_name = event.relation.name
+        remote_app = event.app
+        if remote_app is None:
+            raise RuntimeError("_on_relation_changed was triggered by a broken relation.")
+
+        if not event.relation.data[remote_app]:
+            logger.info(
+                "%s hasn't finished configuring, waiting until relation is changed again.",
```

Comment:
I find this a little vague. Can we rephrase it to something like "waiting until the relation data is populated"?

Reply:
Updated, thanks!